openssl-sys/build.rs

```diff
@@ -63,18 +63,24 @@ fn main() {
     println!("cargo:rustc-link-search=native={}", lib_dir.to_string_lossy());
     println!("cargo:include={}", include_dir.to_string_lossy());
 
-    let version = validate_headers(&[include_dir.clone().into()]);
-
-    let libs = match version {
-        Version::Openssl101 | Version::Openssl102 if target.contains("windows") => {
-            ["ssleay32", "libeay32"]
+    let libs_env = env::var("OPENSSL_LIBS").ok();
+    let libs = match libs_env {
+        Some(ref v) => v.split(":").collect(),
+        None => {
+            let version = validate_headers(&[include_dir.clone().into()]);
```

Comment:
This needs to be run unconditionally since it also discovers the build configuration.